arch/rv32i/src/pmp.rs

```diff
@@ -1,25 +1,131 @@
 //! Implementation of the physical memory protection unit (PMP).
 
+use core::cmp;
 use core::fmt;
 
 use crate::csr;
 use kernel;
+use kernel::common::math;
+use kernel::common::registers::register_bitfields;
 use kernel::mpu;
 
-/// Struct storing configuration for a RISCV PMP region.
-/// In accordance with the priviliged ISA 1.10
-/// https://content.riscv.org/wp-content/uploads/2017/05/riscv-privileged-v1.10.pdf
+// Generic PMP config
+register_bitfields![u32,
+    pub pmpcfg [
+        r OFFSET(0) NUMBITS(1) [],
+        w OFFSET(1) NUMBITS(1) [],
+        x OFFSET(2) NUMBITS(1) [],
+        a OFFSET(3) NUMBITS(2) [
+            OFF = 0,
+            TOR = 1,
+            NA4 = 2,
+            NAPOT = 3
+        ],
+        l OFFSET(5) NUMBITS(1) []
+    ]
+];
```

Comment:
This actually ends up being a messy way to store the data as we can't just write it out to the registers: https://github.com/tock/tock/pull/1591/files#diff-f3ecfe08af10a313e851d65df3ac5c34R479

Any thoughts on a better way?

(NOTE: the offsets are wrong here as well)